logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/featureConfig/GetAndSaveFileSharingStatusUseCase.kt

```diff
@@ -0,0 +1,51 @@
+package com.wire.kalium.logic.feature.featureConfig
+
+import com.wire.kalium.logic.NetworkFailure
+import com.wire.kalium.logic.configuration.UserConfigRepository
+import com.wire.kalium.logic.data.featureConfig.FeatureConfigRepository
+import com.wire.kalium.logic.data.featureConfig.FileSharingModel
+import com.wire.kalium.logic.functional.fold
+import com.wire.kalium.network.exceptions.KaliumException
+import com.wire.kalium.network.exceptions.isNoTeam
+
+interface GetAndSaveFileSharingStatusUseCase {
+    suspend operator fun invoke(): GetFileSharingStatusResult
+}
+
+class GetFileSharingStatusUseCaseImpl(
+    private val userConfigRepository: UserConfigRepository,
+    private val featureConfigRepository: FeatureConfigRepository
+) : GetAndSaveFileSharingStatusUseCase {
+    override suspend operator fun invoke(): GetFileSharingStatusResult =
+        featureConfigRepository.getFileSharingFeatureConfig().fold({
+            if (
+                it is NetworkFailure.ServerMiscommunication &&
+                it.kaliumException is KaliumException.InvalidRequestError
+            ) {
+                if (it.kaliumException.isNoTeam()) {
+                    GetFileSharingStatusResult.Failure.NoTeam
+                } else {
+                    GetFileSharingStatusResult.Failure.OperationDenied
+                }
+            } else {
+                GetFileSharingStatusResult.Failure.Generic(it)
+            }
+
+        }, {
+            if (it.status.lowercase() == "enabled") {
+                userConfigRepository.persistFileSharingStatus(true)
+            } else {
+                userConfigRepository.persistFileSharingStatus(false)
```

Comment:
also, you can remove the if and replace the boolean value with the condition like
` userConfigRepository.persistFileSharingStatus(it.status.lowercase() == "enabled")`